environments/jvm/src/main/java/io/fission/Server.java

```diff
@@ -0,0 +1,122 @@
+package io.fission;
+
+import java.io.File;
+import java.io.IOException;
+import java.net.MalformedURLException;
+import java.net.URL;
+import java.net.URLClassLoader;
+import java.util.Enumeration;
+import java.util.jar.JarEntry;
+import java.util.jar.JarFile;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+import org.springframework.boot.*;
+import org.springframework.boot.autoconfigure.*;
+import org.springframework.http.RequestEntity;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.*;
+
+import io.fission.Function;
+
+@RestController
+@EnableAutoConfiguration
+public class Server {
+		
+	private Function fn;
+
+	private static final int CLASS_LENGTH = 6;
+	
+	private static Logger logger = Logger.getGlobal();
+	
+	@RequestMapping(value = "/", method = { RequestMethod.GET, RequestMethod.POST, RequestMethod.DELETE, RequestMethod.PUT })
+	ResponseEntity<Object> home(RequestEntity<?> req){
+		if (fn == null ) {
+			return ResponseEntity.badRequest().body("Container not specialized");
+		} else {
+			return ((ResponseEntity<Object>) ((Function) fn).call(req, null));
+		}
+	}	
+
+    @PostMapping(path = "/v2/specialize", consumes = "application/json")
+    ResponseEntity<String> specialize(@RequestBody FunctionLoadRequest req){
+        long startTime = System.nanoTime();
+    	File file = new File(req.getFilepath());
+    	if (!file.exists()) {
+    		return ResponseEntity.badRequest().body("/userfunc/user not found");
+    	}
+    	
+    	String entryPoint = req.getFunctionName();
+    	logger.log(Level.INFO, "Entrypoint class:"+ entryPoint);
+    	if (entryPoint == null) {
+    		return ResponseEntity.badRequest().body("Entrypoint class is missing in the function");
+    	}
+
+    	JarFile jarFile = null;
+    	ClassLoader cl = null;
+    	try {
+    		
+    		jarFile = new JarFile(file);
+    		Enumeration<JarEntry> e = jarFile.entries();
+    		URL[] urls = { new URL("jar:file:" + file+"!/") };
+
+    		//TODO Check if the classloading can be improved for ex. use something like: 
+    		// Thread.currentThread().setContextClassLoader(cl);
+    		if (this.getClass().getClassLoader() == null) {
+    			cl = URLClassLoader.newInstance(urls);	
+    		} else {
+    			cl = URLClassLoader.newInstance(urls, this.getClass().getClassLoader());
+    		}
+    		
+    		if (cl == null) {
+    			return ResponseEntity.status(500).body("Failed to initialize the classloader");
+    		}
+    		
+    		// Load all dependent classes from libraries etc. 
+    		while (e.hasMoreElements()) {
+    		    JarEntry je = e.nextElement();
+    		    if(je.isDirectory() || !je.getName().endsWith(".class")){
```

Comment:
1) So I am not 100% sure how packaged Java applications are structured internally, but are you sure that this won't lead to problems with more complex applications (that might have nested structures)?
2) Resources in the package are not added to the classpath. So if the function interacts with (for example) some config or data file in its own classpath, it will break here.

I think both issues are not relevant to simple use cases/functions. So I would be in favor of adding these potential issues to GH issues, instead of stalling this PR.

Reply:
On (1) - I am not entirely sure but I have seen most people using JarEntry to load classes, which I will check is safe in terms of nested & dependent classes.

on (2) - the recent convention is not to use the properties file but put such properties in Java classes. The properties which change often - supply from outside such as a environment variable/configmap or similar such service.

Reply:
For (2), okay forget I said configs :) What about data files or any other resources that are packaged into the JAR?

Reply:
I agree that there can be resources such as SQL file in case of Liquibase and such, but I would take them for specific use cases.